packages/framework/src/Component/EntityLog/ChangeSet/DataTypeResolver/DateTimeDataTypeResolver.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Component\EntityLog\ChangeSet\DataTypeResolver;
+
+use DateTime;
+use DateTimeImmutable;
+use Shopsys\FrameworkBundle\Component\EntityLog\ChangeSet\ResolvedChanges;
+use Shopsys\FrameworkBundle\Component\EntityLog\Model\EntityLogFacade;
+
+class DateTimeDataTypeResolver extends AbstractDataTypeResolver
+{
+    protected const DATE_TIME_FORMAT_WITH_TIMEZONE = 'c';
+    protected const DATE_TIME_FORMAT_FOR_HUMAN = 'd. m. Y H:i:s';
+
+    /**
+     * @param mixed $value
+     * @return bool
+     */
+    protected function isResolvedDataType(mixed $value): bool
+    {
+        return $value instanceof DateTime || $value instanceof DateTimeImmutable;
+    }
+
+    /**
+     * @param array $changes
+     * @return \Shopsys\FrameworkBundle\Component\EntityLog\ChangeSet\ResolvedChanges
+     */
+    public function getResolvedChanges(array $changes): ResolvedChanges
+    {
+        /** @var \DateTimeImmutable|\DateTime|null $oldDateTime */
```

Comment:
It can be written as `@param arrra{0: \DateTimeInterface|null, 1: \DateTimeInterface|null} $changes`